### PR TITLE
Enha: Added product count for each language on vendor dashboad

### PR DESCRIPTION
**Pull request by shahadat015**
WooCommerce WPML handles product translation by creating a product for each language. For that reason, the Dokan vendor dashboard shows product count for all, online and in stock including translated products. I have added also counts for each language on the vendor dashboard.

You have to add this line before end `</ul>` tag of file `dokan-lite/templates/products/listing-status-filter.php`
`<?php do_action( 'dokan_after_product_listing_status_filter' ); ?>`

I have recorded a [short video](https://drive.google.com/file/d/1FR70RgWdOcRbj4xH2HYueUfIAjqbZ9dg/view?usp=sharing). For my device problem voice was not recorded. I was shown how products count shows in the admin > products section. Then the vendor dashboard that I added. The problem was described here.

Fixes [#1366](https://github.com/weDevsOfficial/dokan/issues/1366)

### CODE REVIEW

File: Dokan_Wpml_Products_Count.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Products Count class
+ *
+ * @since 1.0.5
```
Comment (alamgircsebd) [inline Dokan_Wpml_Products_Count.php:5]: use since `1.0.7`

File: Dokan_Wpml_Products_Count.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * @class Dokan_Wpml_Products_Count handle products count for each language
+ */
+class Dokan_Wpml_Products_Count {
+
+    /**
+     * Constructor for Dokan_Wpml_Products_Count class
```
Comment (alamgircsebd) [inline Dokan_Wpml_Products_Count.php:12]: use since

File: Dokan_Wpml_Products_Count.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    public function __construct() {
+        add_action( 'dokan_after_product_listing_status_filter', [ $this, 'show_products_count' ] );
+    }
+
+    /**
+     * Language wise product count
+     *
+     * @since 1.0.5
```
Comment (alamgircsebd) [inline Dokan_Wpml_Products_Count.php:21]: use since `1.0.7`

File: Dokan_Wpml_Products_Count.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * Constructor for Dokan_Wpml_Products_Count class
+     */
+    public function __construct() {
+        add_action( 'dokan_after_product_listing_status_filter', [ $this, 'show_products_count' ] );
+    }
+
+    /**
+     * Language wise product count
```
Comment (alamgircsebd) [inline Dokan_Wpml_Products_Count.php:19]: replace by `Language wise products count`

File: dokan-wpml.php [hunk 5]
```diff
@@ -108,6 +111,40 @@ public static function init() {
...
+     * @return void
+     */
+    public function includes() {
+        require __DIR__ . '/Dokan_Wpml_Products_Count.php';
+    }
+
+    /**
+     * Init all the classes
+     *
+     *  @since 1.0.5
```
Comment (alamgircsebd) [inline dokan-wpml.php:140]: check all since above mentioned

**Review by alamgircsebd — CHANGES_REQUESTED** [review]

File: dokan-wpml.php (hunk 5 above)
Comment (shahadat015) [reply dokan-wpml.php]: @alamgircsebd I have added a new commit with request changes.

File: Dokan_Wpml_Products_Count.php [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+        if ( ! in_array( $cache_key, $tracked_cache_keys, true ) ) {
+            $tracked_cache_keys[] = $cache_key;
+            update_option( $cache_group, $tracked_cache_keys );
+        }
+
+        if ( false === $counts ) {
+            $counts = $wpdb->get_results(
+                $wpdb->prepare(
+                    "SELECT language_code, COUNT(posts.ID) AS count FROM wp_icl_translations translations
```
[the remark this answers is not in the document]
Comment (shahadat015) [reply Dokan_Wpml_Products_Count.php]: Added prefix for wp_icl_translations  table.

File: Dokan_Wpml_Products_Count.php [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                ARRAY_A
+            );
+
+            wp_cache_set( $cache_key, $counts, $cache_group, 3600 * 6 );
+        }
+
+        $html      = '';
+        $languages = wpml_get_active_languages();
+
+        foreach ( $counts as $count ) {
```
[the remark this answers is not in the document]
Comment (shahadat015) [reply Dokan_Wpml_Products_Count.php:72]: I recorded a [short video]( https://drive.google.com/file/d/1FR70RgWdOcRbj4xH2HYueUfIAjqbZ9dg/view?usp=sharing). For my device microphone problem voice was not recorded. I was showed how products count shows in the admin > products section. Then the vendor dashboard that I added. The problem was [described here](https://github.com/weDevsOfficial/dokan/issues/1366).

**Comment by shahadat015** [comment]
Don't need to add count now

**Pull request closed without merging**